src/Views/Components/AdminDetailsPage/index.tsx

```diff
@@ -0,0 +1,211 @@
+/**
+ * External Dependencies
+ */
+import {useEffect, useState} from 'react';
+import {JSONSchemaType} from 'ajv';
+import {ajvResolver} from '@hookform/resolvers/ajv';
+import {FormProvider, SubmitHandler, useForm} from 'react-hook-form';
+
+/**
+ * WordPress Dependencies
+ */
+import {__} from '@wordpress/i18n';
+import {useDispatch} from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal Dependencies
+ */
+import {Spinner as GiveSpinner} from '@givewp/components';
+import TabsRouter from './Tabs/Router';
+import TabList from './Tabs/TabList';
+import {BreadcrumbSeparatorIcon, DotsIcons} from './Icons';
+import NotificationPlaceholder from './Notifications';
+import {AdminDetailsPageProps} from './types';
+import styles from './AdminDetailsPage.module.scss';
+import ErrorBoundary from './ErrorBoundary';
+import TabPanels from './Tabs/TabPanels';
+import DefaultPrimaryActionButton from './DefaultPrimaryActionButton';
+
+import './store';
+
+export default function AdminDetailsPage<T extends Record<string, any>>({
+    objectId,
+    objectType,
+    objectTypePlural,
+    useObjectEntityRecord,
+    resetForm,
+    shouldSaveForm,
+    breadcrumbUrl,
+    StatusBadge,
+    PrimaryActionButton = DefaultPrimaryActionButton,
+    SecondaryActionButton,
+    ContextMenuItems,
+    tabDefinitions,
+    children,
+}: AdminDetailsPageProps<T>) {
+    const [resolver, setResolver] = useState({});
+    const [isSaving, setIsSaving] = useState(false);
+    const [showContextMenu, setShowContextMenu] = useState<boolean>(false);
+
+    const dispatch = useDispatch(`givewp/admin-details-page-notifications`);
+
+    useEffect(() => {
+        apiFetch({
+            path: `/givewp/v3/${objectTypePlural}/${objectId}`,
+            method: 'OPTIONS',
+        }).then(({schema}: {schema: JSONSchemaType<any>}) => {
+            setResolver({
+                resolver: ajvResolver(schema),
+            });
+        });
+    }, []);
+
+    const {entity, hasResolved, save, edit} = useObjectEntityRecord(objectId);
+
+    const methods = useForm<T>({
+        mode: 'onBlur',
+        shouldFocusError: true,
+        ...resolver,
+    });
+
+    const {formState, handleSubmit, reset} = methods;
+
+    // Close context menu when clicked outside
+    useEffect(() => {
+        document.addEventListener('click', (e) => {
+            if (showContextMenu) {
+                return;
+            }
+
+            if (
+                e.target instanceof HTMLElement &&
+                !e.target.closest(`.${styles.donorButtonDots}`) &&
+                !e.target.closest(`.${styles.contextMenu}`)
+            ) {
+                setShowContextMenu(false);
```

Comment:
We shouldn't be using `querySelector`, it's inefficient and completely opposite of the declarative way of doing things in React. I'd suggest using `useRef`.